### PR TITLE
Add Typing hint 

### LINKED ISSUE #17: Add Typing Hint
To avoid errors like #15

### PR DESCRIPTION
**Pull request by drgarcia1986**
fix #17 

Maybe y'all notice but I didn't add type annotations on decorator and context-manager of circuit-breakers 'cause is a pain in the ass

### CODE REVIEW

File: lasier/adapters/caches/aiocache.py [hunk 1]
```diff
@@ -1,16 +1,20 @@
+from typing import Optional
+
 from .base import AsyncCacheAdapterBase
 
 
 class Adapter(AsyncCacheAdapterBase):
 
-    async def add(self, key, value, timeout=None):
+    async def add(
+        self, key: str, value: int, timeout: Optional[int] = None
```
Comment (jarussi) [inline lasier/adapters/caches/aiocache.py:9]: it's more of a personal taste .. but I think one parameter per line is more readable.

File: lasier/circuit_breaker/base.py [hunk 2]
```diff
@@ -8,13 +11,13 @@ class CircuitBreakerBase:
...
-        failure_timeout=None,
-        circuit_timeout=None,
-        catch_exceptions=None,
-    ):
+        rule: BaseRule,
+        cache: Any,
+        failure_exception: Type[Exception],
+        failure_timeout: Optional[float] = None,
+        circuit_timeout: Optional[float] = None,
+        catch_exceptions: Optional[Iterable[Type[Exception]]] = None,
```
Comment (jarussi) [inline lasier/circuit_breaker/base.py:19]: here you used one paremeter per line .. maybe we should use only one way to do it

File: lasier/circuit_breaker/rules/base.py [hunk 3]
```diff
@@ -1,22 +1,29 @@
...
+    def __init__(
+        self, failure_cache_key: str, request_cache_key: Optional[str] = None
+    ) -> None:
         self.failure_cache_key = failure_cache_key
         self.request_cache_key = request_cache_key
 
     @abc.abstractmethod
-    def should_open_circuit(self, total_failures, total_requests):
+    def should_open_circuit(
+        self, total_failures: int, total_requests: int
```
Comment (jarussi) [inline lasier/circuit_breaker/rules/base.py:15]: same comment here about one parameter per line

File: lasier/circuit_breaker/rules/base.py [hunk 4]
```diff
@@ -1,22 +1,29 @@
...
         return self.failure_cache_key is not None
 
-    def should_increase_request_count(self):
+    def should_increase_request_count(self) -> bool:
         return self.request_cache_key is not None
 
     @abc.abstractmethod
-    def log_increase_failures(self, total_failures, total_requests):
+    def log_increase_failures(
+        self, total_failures: int, total_requests: int
```
Comment (jarussi) [inline lasier/circuit_breaker/rules/base.py:27]: here too .. one parameter per line

File: lasier/circuit_breaker/rules/max_failures.py [hunk 5]
```diff
@@ -7,16 +7,20 @@
...
 class MaxFailuresRule(BaseRule):
 
-    def __init__(self, max_failures, failure_cache_key):
+    def __init__(self, max_failures: int, failure_cache_key: str) -> None:
         super().__init__(failure_cache_key=failure_cache_key)
         self.max_failures = max_failures
 
-    def should_open_circuit(self, total_failures, total_requests):
+    def should_open_circuit(
+        self, total_failures: int, total_requests: int
```
Comment (jarussi) [inline lasier/circuit_breaker/rules/max_failures.py:15]: here too .. one parameter per line

File: lasier/circuit_breaker/rules/max_failures.py [hunk 6]
```diff
@@ -7,16 +7,20 @@
...
+    ) -> bool:
         return total_failures >= self.max_failures
 
-    def log_increase_failures(self, total_failures, total_requests):
+    def log_increase_failures(
+        self, total_failures: int, total_requests: int
```
Comment (jarussi) [inline lasier/circuit_breaker/rules/max_failures.py:20]: here too .. one parameter per line

**Comment by drgarcia1986** [comment]
Hi @jarussi  thanks for your review.
About _"one parameter per line"_ I was follow the [black](https://github.com/psf/black) style guide ('cause I'll add black on project soon). I really like the "one parameter per line" pattern, but I like much more the idea that I don't to care about code style, the tool can handle this for me

**Comment by drgarcia1986** [comment]
Hey @renanivo and @linuxsoares thank you too guys

**Pull request merged**